src/cli/filter_arg.zig

```diff
@@ -0,0 +1,183 @@
+const std = @import("std");
+const root = @import("root");
+const bun = root.bun;
+const string = bun.string;
+const Output = bun.Output;
+const Global = bun.Global;
+const strings = bun.strings;
+const json_parser = bun.JSON;
+const Glob = @import("../glob.zig");
+
+const Package = @import("../install/lockfile.zig").Package;
+
+fn findWorkspaceMembers(allocator: std.mem.Allocator, log: *bun.logger.Log, workspace_map: *Package.WorkspaceMap, workdir_: []const u8) !void {
+    bun.JSAst.Expr.Data.Store.create(bun.default_allocator);
+    bun.JSAst.Stmt.Data.Store.create(bun.default_allocator);
+
+    defer {
+        bun.JSAst.Expr.Data.Store.reset();
+        bun.JSAst.Stmt.Data.Store.reset();
+    }
+
+    var workdir = workdir_;
+
+    while (true) : (workdir = std.fs.path.dirname(workdir) orelse break) {
+        const parent_trimmed = strings.withoutTrailingSlash(workdir);
+        var buf2: [bun.MAX_PATH_BYTES + 1]u8 = undefined;
```

Comment:
```suggestion
        var buf2: bun.PathBuffer = undefined;
```
is the `+ 1` needed? null byte should be included

Reply:
updated to use PathBuffer